pkg/service/pipeline.go

```diff
@@ -1993,3 +1991,84 @@ func (s *service) ListComponentRuns(ctx context.Context, req *pipelinepb.ListCom
 		PageSize:      int32(pageSize),
 	}, nil
 }
+
+func (s *service) ListPipelineRunsByRequester(ctx context.Context, req *pipelinepb.ListPipelineRunsByCreditOwnerRequest) (*pipelinepb.ListPipelineRunsByCreditOwnerResponse, error) {
+	page := s.pageInRange(req.GetPage())
+	pageSize := s.pageSizeInRange(req.GetPageSize())
+	requesterUID, _ := utils.GetRequesterUIDAndUserUID(ctx)
+
+	log, _ := logger.GetZapLogger(ctx)
+
+	declarations, err := filtering.NewDeclarations([]filtering.DeclarationOption{
+		filtering.DeclareStandardFunctions(),
+		filtering.DeclareIdent("status", filtering.TypeString),
+		filtering.DeclareIdent("source", filtering.TypeString),
+	}...)
+	if err != nil {
+		return nil, err
+	}
+
+	filter, err := filtering.ParseFilter(req, declarations)
+	if err != nil {
+		return nil, err
+	}
+
+	orderBy, err := ordering.ParseOrderBy(req)
+	if err != nil {
+		return nil, err
+	}
+
+	now := time.Now().UTC()
+	startedTimeBegin := time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, time.UTC)
+	if req.GetStart().IsValid() {
+		startedTimeBegin = req.GetStart().AsTime()
+	}
+	startedTimeEnd := now
+	if req.GetStop().IsValid() {
+		startedTimeEnd = req.GetStop().AsTime()
+	}
+
+	if startedTimeBegin.After(startedTimeEnd) {
+		return nil, fmt.Errorf("time range end time %s is earlier than start time %s", startedTimeEnd.Format(time.RFC3339), startedTimeBegin.Format(time.RFC3339))
+	}
+
+	pipelineRuns, totalCount, err := s.repository.GetPaginatedPipelineRunsByRequester(ctx, requesterUID, startedTimeBegin, startedTimeEnd,
```

Comment:
nit: with more than 3 arguments, I'd consider passing a request / params struct rather than a long list of args

Reply:
updated